design/pen.behavior.md

```diff
@@ -0,0 +1,60 @@
+# Pen Behavior
+
+The Pen behavior is a method of recording the path by which a user moves a grabbed object so that various scenearios 
+can be carried out that wish to capture the virtual drawing of a user.  This behavior will carry with it client side visuals
+to ensure low-latency feedback, but will supply the complete curve data needed by the MRE app to acto on once the drawing is done.
+
+## Architecture
+
+Create a new behavior that will enable grabbable on the actor that the behavior is added to.  This behavior will then allow the user
+to execute a primary action to begin recording the transform changes over time to use it to build a spline curve.  The MRE runtime will
+generate a wet ink effect and a mesh to apply to this spline curve.  Callbacks will occur for when the actor with the behavior has been
+picked up and dropped, as well as when the drawing has started and stopped.  There will be transform data sent up to the MRE at a set 
+frequency while the drawing is active to facilitate spline generation by the MRE app.
+
+### Behavior
+``` ts
+/**
+ * Button behavior class containing the target behavior actions.
+ */
+export class PenBehavior extends TargetBehavior {
+	private _hold: DiscreteAction = new DiscreteAction();
+	private _drawing: DiscreteAction = new DiscreteAction();
+
+	/** @inheritdoc */
+	public get behaviorType(): BehaviorType { return 'pen'; }
+
+	public get hold() { return this._hold; }
+	public get drawing() { return this._drawing; }
+
+	public onDrawDataReceived((data: Transform[]) => void);
+```
+
+## Network
+
+The network messages for this behavior will use the standard behavior payloads for actions started and stopped.  The two actions supported 
+by the Pen behavior are hold (begin/end) and drawing (begin/end).
+
+In addition to the standard `PerformAction` payload, there will need to be a second message that is used for sending transform recordings
+up to the MRE app on a set frequeny cadence.  This payload may look something like this:
+
+``` ts
+export type DrawRecording = Payload & {
+	type: 'draw-recording';
+	actorId: drawingActor;
```

Comment:
Is `drawingActor` a typo?

Reply:
I needed a better name.  I think it should be the `toolActorId`.